CMakeLists.txt

```diff
@@ -98,16 +96,22 @@ set(CMAKE_CXX_STANDARD 14)
 # Set up flags and dependencies:
 #----------------------------------------------------------
 
+# Note: any target dependency of arbor needs to be explicitly added
+# to the 'export set', even the private ones, and this must be done
+# in the same CMakeLists.txt in which the target is defined.
+
 # Interface library `arbor-private-deps` collects dependencies, options etc.
 # for the arbor library.
 
 add_library(arbor-private-deps INTERFACE)
+install(TARGETS arbor-private-deps EXPORT arbor-targets)
 
 # Interface library `arbor-public-deps` collects requirements for the
 # users of the arbor library (e.g. mpi) that will become part
 # of arbor's PUBLIC interface.
 
 add_library(arbor-public-deps INTERFACE)
+install(TARGETS arbor-public-deps EXPORT arbor-targets)
```

Comment:
It makes sense that we *actually* want to install these. This will be libraries and headers (and flags?)?

Reply:
These `EXPORT`-only install commands will just end up inserting the interface target in the generated config files; they don't actually install any files anywhere.